### PR TITLE
Vulkan generator wrap up

### PR DESCRIPTION
**Pull request by yalcinmelihyasin**
This is big chunk of work for finalising the Vulkan parser work.

 Each commit has it's own description. Therefore please review each commit separately.

### CODE REVIEW

File: vulkan_generator/vulkan_parser/parser.py [hunk 1]
```diff
@@ -16,11 +16,61 @@
...
 
 from vulkan_generator.vulkan_parser.api import types
-from vulkan_generator.vulkan_parser.internal import parser as vulkan_parser
+from vulkan_generator.vulkan_parser.internal import parser as internal_parser
 from vulkan_generator.vulkan_parser.postprocess import postprocess
 
 
-def parse(filename: Path) -> types.VulkanInfo:
-    return postprocess.process(vulkan_parser.parse(filename))
+def print_vulkan_metadata(vulkan_metadata: types.VulkanInfo) -> None:
```
Comment (purvisa-at-google-com) [inline vulkan_generator/vulkan_parser/parser.py:26]: Could this just be print(vulkan_metadata)? It seems weird we need to call print on every object member individually

File: vulkan_generator/vulkan_parser/parser.py [hunk 2]
```diff
@@ -16,11 +16,61 @@
...
+    if len(sys.argv) == 2:
+        parse(Path(sys.argv[1]))
+        sys.exit(0)
+
+    if len(sys.argv) == 3:
+        if sys.argv[2].strip().lower() == "dump":
+            parse(Path(sys.argv[1]), True)
+            sys.exit(0)
+
+    print("Please use as <xml location> Optional[<dump>]")
```
Comment (purvisa-at-google-com) [inline vulkan_generator/vulkan_parser/parser.py:76]: <xml location> implies "substitute location here". <dump> means "write 'dump' or do not write 'dump'". Maybe you should choose some way of communicating this that shows the difference. I'd suggest Optional['dump']

File: vulkan_generator/vulkan_parser/parser.py (hunk 1 above)
Comment (yalcinmelihyasin) [reply vulkan_generator/vulkan_parser/parser.py:26]: True, It used to be that I was printing specific things. We don't need a separate function anymore.

File: vulkan_generator/vulkan_parser/parser.py (hunk 2 above)
Comment (yalcinmelihyasin) [reply vulkan_generator/vulkan_parser/parser.py:76]: Sure.

File: vulkan_generator/vulkan_parser/api/types.py [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+    aliased_field: VulkanEnumField
+
+    # Is this define added after Vulkan 1.0 via extension or a later core version
+    extension: bool
+
+    parent: VulkanEnum
+
+    # Print parent's typename only to prevent circular dependency
+    def to_str(self) -> str:
+        return f"""
```
Comment (purvisa-at-google-com) [inline vulkan_generator/vulkan_parser/api/types.py:145]: Anywhere you wrap an fstring across multiple lines like this, you should use dedent() to remove the whitespace added from "where the fstring is hosted in the code"

File: vulkan_generator/vulkan_parser/postprocess/postprocess.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module responsible for postprocessing the Vulkan Metadata
+
+All the stringly typed references will be linked during this stage.
```
[the remark this answers is not in the document]
Comment (yalcinmelihyasin) [reply vulkan_generator/vulkan_parser/postprocess/postprocess.py:18]: Aha, I thought it is a common joke in software engineering. But we can add a description.

File: vulkan_generator/vulkan_parser/postprocess/postprocess.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+from vulkan_generator.vulkan_parser.postprocess import vulkan_commands
+from vulkan_generator.vulkan_parser.postprocess import vulkan_image_formats
+from vulkan_generator.vulkan_parser.postprocess import vulkan_includes
+from vulkan_generator.vulkan_parser.postprocess import vulkan_platforms
+from vulkan_generator.vulkan_parser.postprocess import vulkan_types
+from vulkan_generator.vulkan_parser.postprocess import vulkan_versions_and_extensions
+from vulkan_generator.vulkan_parser.api import types
+
+
+def process(internal_metadata: internal_types.VulkanMetadata) -> types.VulkanInfo:
```
[the remark this answers is not in the document]
Comment (yalcinmelihyasin) [reply vulkan_generator/vulkan_parser/postprocess/postprocess.py:34]: So my thinking was XML -> Metadata -> Info, as in metadata is just semi organised data to create an organised info. But if you feel strong about it, we can change.

**Comment by purvisa-at-google-com** [comment]
LGTM

**Review by purvisa-at-google-com — APPROVED** [review]
This is too large to fine-detail review on the relevant timescales, but it looks correct to me. We can backtrack or change anything problematic that comes up later.

**Pull request merged**